### PR TITLE
Fixes #288, #289, inference_repo issue number 1335

### LINKED ISSUE #288: Uncertainty samples are ignored and submission checker is passing
Please see the below logs where we have uncertain samples during the testing phase and still the submission checker is passing. 
```
02-10-2023 22:53:51.677: Go with mark '2023-02-10_22-52-42_testing'
02-10-2023 22:53:51.677: Response to client sent: Starting untimed measurement, maximum 500000 samples at 1000ms with 0 rampup samples
ptd-server 2023-02-10 22:53:51,678 [INFO] Reply from ptd: 'Starting untimed measurement, maximum 500000 samples at 1000ms with 0 rampup samples'
ptd-server 2023-02-10 22:53:51,678 [INFO] Sending reply to client 'OK'
02-10-2023 22:53:51.982: WARNING: Uncertainty 1.45%, which is above 1.00% limit for the last sample!
02-10-2023 22:53:52.982: WARNING: Uncertainty 1.21%, which is above 1.00% limit for the last sample!
ptd-server 2023-02-10 22:54:09,618 [INFO] Got command from the client 'session,2023-02-10_22-52-42,stop,testing'
02-10-2023 22:54:09.982: WARNING: Uncertainty 1.02%, which is above 1.00% limit for the last sample!
02-10-2023 22:54:10.983: WARNING: Uncertainty 1.40%, which is above 1.00% limit for the last sample!
02-10-2023 22:54:11.982: WARNING: Uncertainty 1.38%, which is above 1.00% limit for the last sample!
02-10-2023 22:54:12.983: WARNING: Uncertainty 1.42%, which is above 1.00% limit for the last sample!
02-10-2023 22:54:13.982: WARNING: Uncertainty 1.45%, which is above 1.00% limit for the last sample!
02-10-2023 22:54:14.982: WARNING: Uncertainty 1.45%, which is above 1.00% limit for the last sample!
02-10-2023 22:54:15.981: WARNING: Uncertainty 1.44%, which is above 1.00% limit for the last sample!
02-10-2023 22:54:16.982: WARNING: Uncertainty 1.45%, which is above 1.00% limit for the last sample!
02-10-2023 22:54:17.982: WARNING: Uncertainty 1.40%, which is above 1.00% limit for the last sample!
02-10-2023 22:54:18.983: WARNING: Uncertainty 1.42%, which is above 1.00% limit for the last sample!
```

```
[x] Check client sources checksum
[x] Check server sources checksum
[x] Check PTD commands and replies
[x] Check UUID
[x] Check session name
[x] Check time difference
[x] Check client server messages
[x] Check results checksum
[x] Check errors and warnings from PTD logs
	'02-10-2023 22:53:11.862: ERROR: Bad watts reading nan from WT310' in ptd_log.txt during ranging stage. Treated as WARNING

[x] Check PTD configuration
[x] Check debug is disabled on server-side

All checks passed. Warnings encountered, check for audit!
```

### LINKED ISSUE #289: Is this a typo in the checker?
[This](https://github.com/mlcommons/power-dev/blob/master/compliance/check.py#L531) seems a typo which needs to be fixed. I was seeing the code to give a fix to [this issue](https://github.com/mlcommons/inference/issues/1335)

### PR DESCRIPTION
**Pull request by arjunsuresh**

### CODE REVIEW

**Comment by arjunsuresh** [comment]
I ran the checker in the inference_results_3.0 repository and below are the results.

```
[2023-05-01 22:28:36,662 submission_checker.py:2651 INFO] ---
[2023-05-01 22:28:36,665 submission_checker.py:2654 ERROR] NoResults closed/Krai/results/firefly-tflite-v2.11.0-ruy/resnet50
[2023-05-01 22:28:36,665 submission_checker.py:2654 ERROR] NoResults closed/Krai/results/firefly-tflite-v2.11.0-ruy/resnet50/multistream
[2023-05-01 22:28:36,665 submission_checker.py:2654 ERROR] NoResults closed/Krai/results/firefly-tflite-v2.11.0-ruy/resnet50/offline
[2023-05-01 22:28:36,665 submission_checker.py:2654 ERROR] NoResults closed/Krai/results/firefly-tflite-v2.11.0-ruy/resnet50/singlestream
[2023-05-01 22:28:36,665 submission_checker.py:2654 ERROR] NoResults open/Krai/results/firefly-tflite-v2.11.0-ruy/mobilenet-v1-1.0-128-non-quantized/multistream
[2023-05-01 22:28:36,665 submission_checker.py:2654 ERROR] NoResults open/Krai/results/firefly-tflite-v2.11.0-ruy/mobilenet-v1-1.0-128-non-quantized/offline
[2023-05-01 22:28:36,666 submission_checker.py:2654 ERROR] NoResults open/Krai/results/firefly-tflite-v2.11.0-ruy/mobilenet-v1-1.0-128-non-quantized/singlestream
[2023-05-01 22:28:36,666 submission_checker.py:2657 INFO] ---
[2023-05-01 22:28:36,666 submission_checker.py:2658 INFO] Results=7277, NoResults=7
[2023-05-01 22:28:36,666 submission_checker.py:2661 ERROR] SUMMARY: submission has errors
```

Even though 7 results are failed it is actually 2 unique results (others are inferred). 

* [First log](https://github.com/mlcommons/inference_results_v3.0/blob/main/closed/Krai/results/firefly-tflite-v2.11.0-ruy/resnet50/singlestream/performance/power/ptd_logs.txt) - here the testing phase current usage is likely above the set range of 0.2A and so this is a good catch by the check.

* [Second log](https://github.com/mlcommons/inference_results_v3.0/blob/main/closed/Krai/results/firefly-tflite-v2.11.0-ruy/resnet50/singlestream/performance/power/ptd_logs.txt)

Both are from the same SUT and the uncertainties are happening at the beginning and end of the loadgen testing phase run.

**Comment by dmiskovic-NV** [comment]
TL;DR: If range is kept constant, lower power will have higher uncertainty

Uncertainty is sum of uncertainties that come from: set range for voltage and current (which are not scaled by measured power), measured value, power factor, and few more parasitic effects. In order to get uncertainty in %, value is divided by measured power, so it will naturally increase as power reduces

**Comment by arjunsuresh** [comment]
Thank you @psyhtest for sharing the details. Actually there [is a check](https://github.com/GATEOverflow/power-dev/blob/master/compliance/check.py#L550) to ensure the uncertainty reports are only considered during the loadgen run. And it is just one single sample which is failing this test for both the SUTs. 

If I modify the check as follows
`if start_load_time+TIME_DELTA_TOLERANCE < log_time < stop_load_time-TIME_DELTA_TOLERANCE:`

it passes and the TIME_DELTA_TOLERANCE being used is 500ms. Would you recommend committing this change?

**Comment by psyhtest** [comment]
> And it is just one single sample which is failing this test for both the SUTs.

Interesting. Where does this sample occur? When transitioning from idle to busy or vice versa, I guess?

**Comment by arjunsuresh** [comment]
yes @psyhtest it occured very close to the testing start - within 500ms interval. Just a guess - this could be due to [this issue](https://github.com/mlcommons/power-dev/issues/236)

**Review by araghun — APPROVED** [review]
Approved in PowerWG 5/16. Pending item post merge is have additional documentation for prove changes.
